classes/sushi/CounterR5Report.inc.php

```diff
@@ -0,0 +1,537 @@
+<?php
+
+/**
+* @file classes/sushi/CounterR5Report.inc.php
+*
+* Copyright (c) 2013-2021 Simon Fraser University
+* Copyright (c) 2003-2021 John Willinsky
+* Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
+*
+* @class CounterR5Report
+* @ingroup sushi
+*
+* @brief Base class for COUNTER R5 reports
+*
+*/
+
+namespace PKP\sushi;
+
+use APP\core\Services;
+use APP\facades\Repo;
+use DateTime;
+use PKP\context\Context;
+
+abstract class CounterR5Report
+{
+    /** The access method */
+    public const ACCESS_METHOD = 'Regular';
+
+    /** Access type */
+    public const ACCESS_TYPE = 'OA_Gold';
```

Comment:
Is this accurate for all users?

Reply:
COUNTER R5 currently considers two access types, controlled and oa_gold. OA_Gold is only if the content has **always** been open. Currently, controlled is also content that is OA after some/embargo period. Thus we would need to track that at the submission level. S. https://github.com/pkp/pkp-lib/issues/6782#issuecomment-858610664.
Thus, we decided to 'ignore' this filter ACCESS_TYPE at the moment, s.  this https://github.com/pkp/pkp-lib/issues/6782#issuecomment-858621390. I.e. if subscription journals would need to have a better differentiation they would need to implement that tracking on submission level. For now we only support OA.

Reply:
Could the report maybe then be disabled when access is configured not to be open access? I'm on the fence about whether this is worth the trouble, but it seems like it should be impossible to generate a report when the access mode is unsupported, and we do have that setting ready to hand.